policy-controller/src/main.rs

```diff
@@ -133,9 +139,58 @@ async fn main() -> Result<()> {
         probe_networks,
     });
 
+    // Fetch the policy-controller deployment so that we can use it as an owner
+    // reference of the Lease.
+    let api = k8s::Api::<Deployment>::namespaced(runtime.client(), &control_plane_namespace);
+    let deployment = api.get(&policy_deployment_name).await?;
+
+    let api = k8s::Api::namespaced(runtime.client(), &control_plane_namespace);
+    let params = PatchParams {
+        field_manager: Some("policy-controller".to_string()),
+        ..Default::default()
+    };
+    match api
+        .patch(
+            LEASE_NAME,
+            &params,
+            &kube::api::Patch::Apply(coordv1::Lease {
+                metadata: ObjectMeta {
+                    name: Some(LEASE_NAME.to_string()),
+                    namespace: Some(control_plane_namespace.clone()),
+                    // Specifying a resource version of "0" means that we will
+                    // only create the Lease if it does not already exist.
+                    resource_version: Some("0".to_string()),
+                    owner_references: Some(vec![deployment.controller_owner_ref(&()).unwrap()]),
+                    labels: Some(
+                        [
+                            (
+                                "linkerd.io/control-plane-component".to_string(),
+                                "destination".to_string(),
+                            ),
+                            (
+                                "linkerd.io/control-plane-ns".to_string(),
+                                control_plane_namespace,
+                            ),
+                        ]
+                        .into_iter()
+                        .collect(),
+                    ),
+                    ..Default::default()
+                },
+                spec: None,
+            }),
+        )
+        .await
+    {
+        Ok(lease) => tracing::info!(?lease, "created Lease resource"),
+        Err(k8s::Error::Api(_)) => {} // Lease already exists, no need to create it.
+        Err(error) => {
+            tracing::error!(%error, "error creating Lease resource");
+            return Err(error.into());
+        }
+    };
     // Create the lease manager used for trying to claim the policy
     // controller write lease.
-    let api = k8s::Api::namespaced(runtime.client(), &control_plane_namespace);
     // todo: Do we need to use LeaseManager::field_manager here?
     let lease = kubert::lease::LeaseManager::init(api, LEASE_NAME).await?;
```

Comment:
nit, take it or leave it: i _might_ consider sticking all of this code in a `fn init_lease(...) -> Result<LeaseManager>` or something. i don't think we'll ever call it anywhere except in `main`, but IMO the `main` function might read a little better if all of this code was just a call to `init_lease`, which you could then go and read if you actually needed to know what initializing the lease meant? on the other hand, maybe we don't want to make functions for code that isn't actually called more than once, :woman_shrugging:...